src/main/java/appeng/tile/storage/TileChest.java

```diff
@@ -703,6 +710,70 @@ public void saveChanges(final IMEInventory cellInventory) {
         this.worldObj.markTileEntityChunkModified(this.xCoord, this.yCoord, this.zCoord, this);
     }
 
+    public boolean lockCells() {
+        final ItemStack cell = this.inv.getStackInSlot(1);
+        if (cellHandler == null || cell == null
+                || !(cell.getItem() instanceof ItemExtremeStorageCell)
+                || (cell.getItem() instanceof ItemExtremeStorageCell exCell && exCell.getTotalTypes(cell) != 1)) {
+            return true;
+        }
+        final IMEInventoryHandler<?> inv = cellHandler.getCellInventory(cell, this, StorageChannel.ITEMS);
+        if (inv instanceof ICellInventoryHandler handler) {
+            final ICellInventory cellInventory = handler.getCellInv();
+            if (cellInventory != null) {
+                if (cellInventory.getStoredItemTypes() != 0) {
+                    cellInventory.getConfigInventory().setInventorySlotContents(
+                            0,
+                            handler.getAvailableItems(new ItemList()).getFirstItem().getItemStack());
+                }
+            }
+        }
+        return true;
+    }
+
+    public boolean applyStickyToCells(EntityPlayer p) {
+        ItemStack cell = this.inv.getStackInSlot(1);
+        if (cellHandler == null || cell == null
+                || !(cell.getItem() instanceof ItemExtremeStorageCell)
+                || (cell.getItem() instanceof ItemExtremeStorageCell exCell && exCell.getTotalTypes(cell) != 1)) {
+            return true;
+        }
+        if (cell.getItem() instanceof ICellWorkbenchItem cellItem) {
+            final IMEInventoryHandler<?> inv = cellHandler.getCellInventory(cell, this, StorageChannel.ITEMS);
+            if (inv instanceof ICellInventoryHandler handler) {
+                final ICellInventory cellInventory = handler.getCellInv();
+                if (cellInventory != null && cellInventory.getStoredItemTypes() != 0
```

Comment:
Be more in line with the `getTotalTypes` check above
```suggestion
                if (cellInventory != null && cellInventory.getStoredItemTypes() == 1
```

Reply:
Resolved, see d64a646669a6487c86b053665044dae45a2d05be